usr/actor.c

```diff
@@ -168,8 +171,9 @@ actor_schedule_private(actor_t *thread, uint32_t delay_secs, int head)
 		// don't do anything
 		break;
 	case ACTOR_INVALID:
-		log_error("BUG: Trying to schedule a thread that has not been "
-			  "setup. Ignoring sched.");
+		log_error("BUG: Trying to schedule a thread %p:%s"
+			  "that has not been setup. Ignoring sched.\n",
+			  thread, thread->name);
```

Comment:
In this case, it's checking for a schedule call on an actor_t that hasn't been initialized with actor_init (but apparently trusting that actor_t is zeroed so that state == ACTOR_INVALID == 0).  I don't know that it makes sense to trust thread->name in that case.  If the entire struct is zeroed it wouldn't matter, but it's never going to be trustworthy.

Reply:
> In this case, it's checking for a schedule call on an actor_t that hasn't been initialized with actor_init (but apparently trusting that actor_t is zeroed so that state == ACTOR_INVALID == 0). I don't know that it makes sense to trust thread->name in that case. If the entire struct is zeroed it wouldn't matter, but it's never going to be trustworthy.

Yes, here should not use thread->name. I would modify here to print thread address only.